example/lib/main.dart

```diff
@@ -42,88 +52,93 @@ class _MyHomePageState extends State<MyHomePage> {
   UhstSocket? client;
   final TextEditingController _hostIdController = TextEditingController();
 
-  @override
-  void initState() {
-    super.initState();
-  }
-
-  Future<void> initHost() async {
+  Future<void> connectHost() async {
     initUHST();
     host?.disconnect();
     host = uhst?.host();
     host
       ?..onReady(handler: ({required hostId}) {
-        setState(() {
-          hostMessages.add('Host Ready! Using $hostId');
-          print('host is ready!');
-          _hostIdController.text = hostId;
-        });
+        hostMessages.add('Host Ready! Using hostId $hostId');
+        print('host is ready!');
+        _hostIdController.text = hostId;
+        setState(() {});
       })
       ..onException(handler: ({required dynamic exception}) {
-        print('exception received! $exception');
-        setState(() {
-          hostMessages.add(exception.toString());
-        });
+        if (exception is RelayException) {
+          hostMessages.add('disconneted! $exception');
+        } else {
+          hostMessages.add('exception received! $exception');
+        }
+        setState(() {});
+      })
+      ..onClose(handler: ({required hostId}) {
```

Comment:
why pass hostId here? There can be only one host at a time, in onReady we pass the hostId because the relay will assign it, but in onClose why do we need hostId?

Reply:
the reason is that as onClose will be emitted to event stream, and in the moment when listener will be called there will be no hostId in socket as it will be closed - and it will be null

Reply:
But why is the hostId needed when the host is already gone?

Reply:
fixed